Core/GDCore/Project/Layout.cpp

```diff
@@ -505,18 +505,16 @@ gd::String GD_CORE_API GetTypeOfBehavior(const gd::ObjectsContainer& project,
                                          gd::String name,
                                          bool searchInGroups) {
   for (std::size_t i = 0; i < layout.GetObjectsCount(); ++i) {
-    vector<gd::String> behaviors = layout.GetObject(i).GetAllBehaviorNames();
-    for (std::size_t j = 0; j < behaviors.size(); ++j) {
-      if (layout.GetObject(i).GetBehavior(behaviors[j]).GetName() == name)
-        return layout.GetObject(i).GetBehavior(behaviors[j]).GetTypeName();
+    const auto &object = layout.GetObject(i);
+    if (object.HasBehaviorNamed(name)) {
+      return object.GetBehavior(name).GetTypeName();

```

Comment:
`GetAllBehaviorNames()` allocate a vector. As the new type checking access the metadata more often, it made the code generation slower. I replaced it with `HasBehaviorNamed()` but it does the same `find` call as `GetBehavior`. I preferred to leave things unoptimized for code clarity (it would only have saved 1ms on a 2.8 second long export).

Reply:
My god this was so un-optimised 😅
HasBehaviorNamed + GetBehavior is fine :) We at least avoid the unnecessary copy.

I know there are other stuff like the extension/instruction metadata that are doing a lot of useless loops too. We could take a look at converting these to hashtable/unordered_map or something that helps getting constant time accesses :)

Reply:
Yes, I did some other similar optimizations. I don't know if I've changed them all but at least the ones I found with the monitoring tool.